### PR TITLE
transport: fetch and store server certificate

### PR DESCRIPTION
**Pull request by dpwiz**

### CODE REVIEW

File: src/Simplex/Messaging/Crypto.hs [hunk 1]
```diff
@@ -1384,3 +1405,9 @@ keyError :: (a, [ASN1]) -> Either String b
 keyError = \case
   (_, []) -> Left "unknown key algorithm"
   _ -> Left "more than one key"
+
+x509ToPrivateSign :: PrivKey -> Either String APrivateSignKey
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Crypto.hs:1409]: this seems unnecessary, there is already x509ToPrivate, it can be combined with type check

Comment (epoberezkin) [reply src/Simplex/Messaging/Crypto.hs:1409]: with CryptoPrivateKey instance

File: src/Simplex/Messaging/Transport.hs [hunk 2]
```diff
@@ -194,13 +201,25 @@ data TProxy c = TProxy
 
 data ATransport = forall c. Transport c => ATransport (TProxy c)
 
+getServerVerifyKey :: Transport c => c -> Either String C.APublicVerifyKey
+getServerVerifyKey c =
+  case getServerCerts c of
+    X.CertificateChain (server : _ca) -> fromX509 . X.certPubKey . X.signedObject $ X.getSigned server
+    _ -> Left "no certificate chain"
+  where
+    fromX509 = \case
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Transport.hs:210]: this is also like x509ToPublic with type check using CryptoPublicKey

File: src/Simplex/Messaging/Transport.hs [hunk 3]
```diff
@@ -325,30 +348,45 @@ data ClientHandshake = ClientHandshake
...
 
 instance Encoding ServerHandshake where
-  smpEncode ServerHandshake {smpVersionRange, sessionId, authPubKey} =
-    smpEncode (smpVersionRange, sessionId) <> encodeAuthPubKey (maxVersion smpVersionRange) authPubKey
+  smpEncode ServerHandshake {smpVersionRange, sessionId, authPubKey, certChain} =
+    B.concat
+      [ smpEncode (smpVersionRange, sessionId),
+        encodeAuthEncryptCmds maxV $ encodeChain <$> certChain,
+        encodeAuthEncryptCmds maxV $ C.SignedObject <$> authPubKey
+      ]
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Transport.hs:364]: you cannot make it independently of the version, it won't work with the old clients who will only be able to parse smpVersionRange and sessionId

Comment (epoberezkin) [reply src/Simplex/Messaging/Transport.hs:364]: ah, this checks version twice... this seems unnecessarily confusing, needs to change somehow

Comment (dpwiz) [reply src/Simplex/Messaging/Transport.hs:364]: I thought maybe we will add chain and pubkey in different versions. Better put them under one check ofc, if possible.

**Pull request merged**